x-pack/filebeat/input/azureeventhub/input.go

```diff
@@ -167,6 +167,10 @@ func (a *azureInput) processEvents(event *eventhub.Event, partitionID string) bo
 func (a *azureInput) parseMultipleMessages(bMessage []byte) []string {
 	var mapObject map[string][]interface{}
 	var messages []string
+
+	// clean up the message for known issues producing a malformed JSON
+	bMessage = sanitize(bMessage, a.config.SanitizeOptions...)
```

Comment:
As suggested by @gpop63, what about checking for JSON validity using https://pkg.go.dev/encoding/json#Valid before sanitizing the message?

If sanitization is enabled (options are not empty), we check if the document is valid; if is not valid, the we attempt to sanitize it.

WDYT?

Reply:
This approach is efficient if many valid messages are processed while sanitization is enabled, which I expect would be the majority of cases.

An if statement is enough here, since exceptions are evaluated in order and the subsequent expressions are not evaluated if one is already false.